ci-bench/Cargo.toml

```diff
@@ -0,0 +1,22 @@
+[package]
+name = "rustls-ci-bench"
+version = "0.0.1"
+edition = "2021"
+rust-version = "1.60"
+license = "Apache-2.0 OR ISC OR MIT"
+description = "Rustls CPU instruction counting benchmarks."
+publish = false
+
+# Prevent this from interfering with workspaces
```

Comment:
Why don't we want this in the workspace?

Reply:
Hmm I actually copied this from [connect-tests](https://github.com/rustls/rustls/blob/6bdaf04e7af97821485da115b5436913fedea874/connect-tests/Cargo.toml). I haven't worked much with workspaces in the past, and I assumed this was necessary for "infrastructure" code. Is it superfluous?

Reply:
I would propose we ditch it and make this crate part of the workspace, so that it's included in CI by default.

Reply:
All right!

Reply:
The connect tests are a special case because they're hitting 3rd party servers over the internet and so can be prone to flakes outside of our control due to bad internet weather, cert expiry, etc.

Reply:
Hmmm... One drawback of this is that the CI now enforces that the benchmarking code compiles with the minimum supported version. Or is [this](https://github.com/rustls/rustls/actions/runs/5891298366/job/15979135819?pr=1404) an unrelated failure? It also seems to skew the coverage measurements (and I assume has an impact on the length of some CI checks).

What is the benefit of having the new crate be part of the workspace? We are going to add a dedicated CI job for it anyway, so right now I can only see the drawbacks.

Reply:
That failure looks unrelated. And you actually set `rust-version` in the Cargo metadata for this new crate, is that a lie?

Reply:
Hmm I should have looked more closely when I copy-pasted the original `Cargo.toml` file. My natural choice would be to have no minimum rust version for this kind of code, assuming that developers of the library will always be using an up-to-date compiler. Is that ok? Or are there reasons to have a minimum rust version even for code that is not part of the library?

Reply:
Actually the MSRV workflow in CI is already targeted at `-p rustls` only, so there should be no problem avoiding specifying a `rust-version` here and just using the stable compiler as used for most of the CI jobs.

Reply:
All right, I have removed the `rust-version` line from `Cargo.toml`